### ISSUE TITLE
ESLint not able to resolve import - `Error: unable to resolve path to module "express-rate-limit"`

### ISSUE THREAD

**Issue by Phoenix801** [nfriedly/express-rate-limit#277]
### Description

I installed it with:

```
npm install express-rate-limit
```

and imported it as:

```js
const rateLimit = require('express-rate-limit');
```

I'm using ESLint and it errors on the fact that there is no "/node_modules/express-rate-limit/dist/index" module. ESLint probably looks for the plain javascript format but there is no `index.js`, only `index.cjs`, `index.d.ts` and `index.mjs`.

![image](https://user-images.githubusercontent.com/5101061/147831599-a1e9ab9f-6515-4908-b96a-b1de899b4d3f.png)

### Library version

6.0.3

### Node version

14.13.0

### Module system

CommonJS

**Comment by gamemaker1** [comment]
Does it give you the same error (cannot resolve module) when you run the program?

**Comment by endbay** [comment]
![image](https://user-images.githubusercontent.com/38842789/147833231-f3d1c835-1a29-4554-a3e9-356ce02b51f1.png)

Having the same issue with PHPStorm and ESLint v8.5 after updating to v6 latest.

**Comment by gamemaker1** [comment]
Hmmm... let me investigate - but at least it doesn't give you errors while running the program right?

**Comment by gamemaker1** [comment]
It looks like ESLint [does not support the `exports` field](https://github.com/import-js/eslint-plugin-import/issues/1868) in package.json - which is what this library uses to dual-export CJS and ESM versions.

**Comment by gamemaker1** [comment]
Could you try adding `"main": "./dist/index.cjs"` to `node_modules/express-rate-limit/package.json`?

I had removed the `main` field as part of a fix for jest not working with this module - see #267.

**Comment by pgrodrigues** [comment]
> Could you try adding `"main": "./dist/index.cjs"` to `node_modules/express-rate-limit/package.json`?
> 
> I had removed the `main` field as part of a fix for jest not working with this module - see #267.

@gamemaker1 after performing the change you mentioned and running ESLint it no longer complains about the issue.

**Comment by nfriedly** [comment]
I wonder if adding a root level index.js that just `require()`s the CJS module and re-exports it would help?

**Comment by gamemaker1** [comment]
> I wonder if adding a root level index.js that just require()s the CJS module and re-exports it would help?

I'm not sure that would exactly help, because we would still need to add a `main` field to the package.json pointing to the index.js file.

I think yargs [also does this dual export stuff](https://github.com/yargs/yargs/blob/main/package.json), let me see how they do it and apply that to our library.

**Comment by gamemaker1** [comment]
Hi @Phoenix801 @endbay @pgrodrigues,

I [modified](https://github.com/nfriedly/express-rate-limit/commit/5a16de18f1b2642c73384c7492dd641588187dcb) the `exports` field and added the `main` and `module` fields to the library's `package.json`. The import tests still pass, so I think this won't un-fix #267. Could you please try it out as follows:

```
npm install https://raw.githubusercontent.com/gamemaker1/test/express-rate-limit-hotfix/assets/express-rate-limit-6.0.3.tgz
```

...and let me know if this fixes the problem?

Thanks!

**Comment by Phoenix801** [comment]
> Hi @Phoenix801 @endbay @pgrodrigues,
> 
> I [modified](https://github.com/nfriedly/express-rate-limit/commit/5a16de18f1b2642c73384c7492dd641588187dcb) the `exports` field and added the `main` and `module` fields to the library's `package.json`. The import tests still pass, so I think this won't un-fix #267. Could you please try it out as follows:
> 
> ```
> npm install https://raw.githubusercontent.com/gamemaker1/test/express-rate-limit-hotfix/assets/express-rate-limit-6.0.3.tgz
> ```
> 
> ...and let me know if this fixes the problem?
> 
> Thanks!

unfortunately it's already working on a running server and I rather not mess with it.
the fix you gave beforehand with the main property worked.

**Comment by pgrodrigues** [comment]
> Hi @Phoenix801 @endbay @pgrodrigues,
> 
> I [modified](https://github.com/nfriedly/express-rate-limit/commit/5a16de18f1b2642c73384c7492dd641588187dcb) the `exports` field and added the `main` and `module` fields to the library's `package.json`. The import tests still pass, so I think this won't un-fix #267. Could you please try it out as follows:
> 
> ```
> npm install https://raw.githubusercontent.com/gamemaker1/test/express-rate-limit-hotfix/assets/express-rate-limit-6.0.3.tgz
> ```
> 
> ...and let me know if this fixes the problem?
> 
> Thanks!

In my case, no issue is reported from ESLint.

**Comment by gamemaker1** [comment]
Ok I believe the fix works 🎉 and all the tests still pass too, especially the import ones.

@nfriedly should I release a patch version (6.0.4) for this?

**Comment by nfriedly** [comment]
@gamemaker1 yea, probably. Do you mind opening a PR so I can take a look at the changes? 

Also, how hard would it be to replicate the issue in a test? It might be more effort than it's worth, but if you can think of an easy way to do it, it could be good.

**Comment by gamemaker1** [comment]
Ummm... I already committed that change to `master`. I can revert it and put it in as a PR instead maybe?

> Also, how hard would it be to replicate the issue in a test? It might be more effort than it's worth, but if you can think of an easy way to do it, it could be good.

Shouldn't be that hard - we just need to create a new sample project and run eslint in the tests.

**Comment by Phoenix801** [comment]
> npm install https://raw.githubusercontent.com/gamemaker1/test/express-rate-limit-hotfix/assets/express-rate-limit-6.0.3.tgz

installed it on a new project, didn't get any warnings with lint :)

**Comment by nfriedly** [comment]
> Ummm... I already committed that change to master. I can revert it and put it in as a PR instead maybe?

Oh, nevermind, no need to do that. I'd prefer if we got the tests figured out first, but yea, I think it's safe to ship.

**Comment by nfriedly** [comment]
Ok, I just went ahead and published v6.0.4 to npm with @gamemaker1's fix.  Please let us know if that works for everyone.
